synapse/storage/registration.py

```diff
@@ -350,3 +352,27 @@ def _count_users(txn):
 
         ret = yield self.runInteraction("count_users", _count_users)
         defer.returnValue(ret)
+
+    @defer.inlineCallbacks
+    def find_next_generated_user_id(self):
```

Comment:
a doc-comment here could be useful to call out the fact that it will return the first hole in the sequence, rather than being relative to the highest allocation so far.